hubspot-style/src/main/java/com/hubspot/immutables/utils/WireSafeEnum.java

```diff
@@ -86,10 +84,19 @@ public static <T extends Enum<T>> WireSafeEnum<T> of(@Nonnull T value) {
   }
 
   @Nonnull
-  @SuppressWarnings("unchecked")
   public static <T extends Enum<T>> WireSafeEnum<T> fromJson(
       @Nonnull Class<T> enumType,
       @Nonnull String jsonValue
+  ) {
+    return fromJson(enumType, jsonValue, WireSafeEnum::new);
+  }
+
+  @Nonnull
+  @SuppressWarnings("unchecked")
+  private static <T extends Enum<T>> WireSafeEnum<T> fromJson(
+      @Nonnull Class<T> enumType,
+      @Nonnull String jsonValue,
+      BiFunction<Class<T>, String, WireSafeEnum<T>> creator
```

Comment:
So this function is like a fallback when we don't think there's a matching constant?

Reply:
If so, thoughts on naming the variable something like `fallback`?

Reply:
Yea, that's the idea 👍

Reply:
Sure, works for me 👍